apps/web/src/components/AssessmentStepTracker/AssessmentStepTracker.tsx

```diff
@@ -17,29 +18,25 @@ export interface AssessmentStepTrackerProps {
 
 const AssessmentStepTracker = ({ pool }: AssessmentStepTrackerProps) => {
   const intl = useIntl();
-  const steps = groupPoolCandidatesByStep(pool);
+  const steps = unpackMaybes(pool.assessmentSteps);
+  const candidates = unpackMaybes(pool.poolCandidates);
+  const groupedSteps = groupPoolCandidatesByStep(steps, candidates);
 
   return (
     <Board.Root>
-      {Array.from(steps.values()).map(({ step, assessments }, index) => {
-        const assessmentsWithDecision = Array.from(assessments.values());
-        return (
-          <Board.Column key={step.id}>
-            <Board.ColumnHeader
-              prefix={intl.formatMessage(applicationMessages.numberedStep, {
-                stepOrdinal: index + 1,
-              })}
-            >
-              {getLocalizedName(step.title, intl)}
-            </Board.ColumnHeader>
-            <ResultsDetails step={step} results={assessmentsWithDecision} />
-            <AssessmentResults
-              stepType={step.type}
-              results={assessmentsWithDecision}
-            />
-          </Board.Column>
-        );
-      })}
+      {groupedSteps.map(({ step, resultCounts, results }, index) => (
+        <Board.Column key={step.id}>
+          <Board.ColumnHeader
+            prefix={intl.formatMessage(applicationMessages.numberedStep, {
+              stepOrdinal: index + 1,
+            })}
+          >
+            {getLocalizedName(step.title, intl)}
+          </Board.ColumnHeader>
+          <ResultsDetails {...{ resultCounts, step }} />
```

Comment:
I've never seen this syntax in a component before. Is this the same as `<ResultsDetails resultCounts={resultsCounts} step={step} />`?

Reply:
Yup, I can update it to use the more common syntax though. We do use it in places but not very frequently,